bootstrap/bootstrap.sh

```diff
@@ -28,8 +29,54 @@ sudo apt-get update
 sudo apt-get -y dist-upgrade
 
 #----nodejs install
-curl -sL https://deb.nodesource.com/setup_7.x | sudo -E bash -
+curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash -
 sudo apt-get install -y nodejs
 
-#----alsa install
-sudo apt-get install -y alsa-base alsa-utils
+#----install official requirements
+sudo apt-get install -y alsa-base alsa-utils libasound2-dev
+
+#----install missing pigpio in Raspbian Lite (the command npm install pigpio will be exec by package.json)
+sudo apt-get install pigpio
+
+#----install git and download tjbot
+if [ ! -d "${TJBOT_FOLDER}/recipes/conversation" ]; then
+    sudo apt-get install -y git
+
+    rm -Rf /home/pi/git/tjbot
+    git clone https://github.com/ibmtjbot/tjbot.git /home/pi/git/tjbot
+
+    TJBOT_FOLDER='/home/pi/git/tjbot'
+fi
+
+#----install conversation (to install it will resolve tjbotlib and other dependencies)
+cd $TJBOT_FOLDER
+cd recipes/conversation
+echo "path: $PWD"
+
+sudo npm install --unsafe-perm -g node-gyp
```

Comment:
I just ran bootstrap.sh on a freshly installed Raspbian Jessie Lite 2017-04-10. bootstrap.sh was modified to use `npm install` rather than `sudo npm install --unsafe-perm -g node-gyp` and `sudo npm install --unsafe-perm -g readline-sync`.

There's a gist [here](https://gist.github.com/fivdi/4be4f73ac08b9e42d74b71e893db96f0) with all the relevant information:
 * `bootstrap.log` - The output of the installation script
 * `bootstrap.sh` - The modified installation script
 * `npm list` - The output of `npm list` after the installation when called in `~/tjbot/bootstrap/tests` to verify that everything was installed locally
 * `npm list -g` - The output of `npm list -g` after the installation when called in `~/tjbot/bootstrap/tests` to ensure that nothing was installed globally

Note that I don't actually have the appropriate hardware attached to the Pi. The test software doesn't know this most of the time but it does with the camera so there was an error with the camera test. There was also an error with the servo test because I was hitting the keyboard too fast and answered one of the questions before it was asked.

I also verified that calling `sudo npm install --unsafe-perm -g readline-sync` in directory `/home/pi/tjbot/bootstrap/tests` will only install package `readline-sync` globally in `/usr/lib/node_modules` and will not install the current package context (/home/pi/tjbot/bootstrap/tests) globally. This probably means that you installed some of the required packages globally in the past and they are still being used today.